### PR TITLE
Feat: add duniter trust bonus 

### PR DESCRIPTION
**Pull request by developerfred**
##### Description

 [skunkworks/issues/187](https://github.com/gitcoinco/skunkworks/issues/187)

So it was the validation flow

1. The user must have the same userid on Duniter and Gitcoin
2. The user needs to be verified as a member
3. User needs to add their Gitcoin URL on Duniter

with these 3 validations OK my code returns the user as validated in Duniter

Env endpoints
``` 
BMAS_ENDPOINT = "BMAS g1.duniter.org 443"
ES_CORE_ENDPOINT = "ES_CORE_API g1.data.duniter.fr 443"
ES_USER_ENDPOINT = "ES_USER_API g1.data.duniter.fr 443"
```
When a Duniter user has 5 certificates he is automatically verified for 5 years in my implementation I check if he is a member and if the account has the same username as the gitcoin account

the Gitcoin user only needs to post using Duniter's public key to do the validations.

## Trust Duniter
[screen movie](https://youtu.be/CrKTciCpO9g)

Screen Shot 2020-11-13 at 08 44 27
Screen Shot 2020-11-13 at 08 44 38

**I don't have the checks on my duniter profile, but any gitcoin user who has these checks can validate his account.**
Screen Shot 2020-11-11 at 12 45 13
Screen Shot 2020-11-11 at 12 45 06

# User Verify on Duniter
Screen Shot 2020-11-11 at 12 44 48

### CODE REVIEW

**Comment by danlipert** [comment]
@owocki check this out and let us know what you think

**Comment by owocki** [comment]
we need to store the duniter verification in the model per the spec.

@developerfred could you post a short video of the UX?

**Comment by developerfred** [comment]
> we need to store the duniter verification in the model per the spec.
> 
> @developerfred could you post a short video of the UX?

I'll do the screen capture in the morning

**Comment by developerfred** [comment]
@owocki, @octavioamu, @danlipert   I updated the PR with more information and the video

**Comment by developerfred** [comment]
@owocki, @octavioamu, @danlipert updated here and write some tests to improve coverage

**Comment by owocki** [comment]
@developerfred GR9 is next week (deploys on monday), and we'd been keen to to get this in.

can you rebase on the current master branch + test to make sure it looks good?

cc @gitcoinco/engineers

File: app/dashboard/migrations/0160_auto_20201111_0445.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Generated by Django 2.2.4 on 2020-11-11 04:45
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (owocki) [inline app/dashboard/migrations/0160_auto_20201111_0445.py:6]: hmm this migration shouldnt be on the PR i dont think. nothing to do with the logic of this PR

File: app/dashboard/migrations/0162_merge_20201112_0712.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Generated by Django 2.2.4 on 2020-11-12 07:12
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
Comment (owocki) [inline app/dashboard/migrations/0162_merge_20201112_0712.py:6]: same with this one

File: app/dashboard/migrations/0166_merge_20201203_2139.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Generated by Django 2.2.4 on 2020-12-03 21:39
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
Comment (owocki) [inline app/dashboard/migrations/0166_merge_20201203_2139.py:6]: lots of extra migrations i think that dont need included here

File: app/dashboard/views.py [hunk 4]
```diff
@@ -3211,12 +3404,6 @@ def connect_google():
 @login_required
 @require_POST
 def request_verify_google(request, handle):
-    is_logged_in_user = request.user.is_authenticated and request.user.username.lower() == handle.lower()
```
Comment (owocki) [inline app/dashboard/views.py:3214]: did you mean to remove this?

File: requirements/base.txt [hunk 5]
```diff
@@ -104,4 +104,11 @@ ed25519
 tweepy
 oogway==0.7.0
 icalendar==4.0.7
+duniterpy
```
Comment (owocki) [inline requirements/base.txt:107]: a lot of new requirements.  are these all required?

File: app/marketing/migrations/0029_merge_20201112_0711.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Generated by Django 2.2.4 on 2020-11-12 07:11
+
+from django.db import migrations
+
```
Comment (owocki) [inline app/marketing/migrations/0029_merge_20201112_0711.py:4]: another one that could maybe be pruned

**Comment by owocki** [comment]
also getting travis to run would be nice

**Comment by owocki** [comment]
for some reason i can't click this button.  not sure whats supposed to happen when i do click it? https://share.getcloudapp.com/DOu20KJY

**Comment by owocki** [comment]
integration branch => https://github.com/gitcoinco/web/pull/8569

**Comment by developerfred** [comment]
> for some reason i can't click this button. not sure whats supposed to happen when i do click it? https://share.getcloudapp.com/DOu20KJY

I will look at that error today. I believe that we need to put a json with a list of addresses from the Duniter API.

**Comment by developerfred** [comment]
@owocki I have a space problem on my macbook, I'm buying a new external hard drive that will arrive on the 18th. I hope to be able to work on the project as soon as the hard drive arrives.

**Comment by owocki** [comment]
ok @developerfred let me know

**Comment by owocki** [comment]
when i try to instal the pip requirement

```
Building wheels for collected packages: pyPEG2, pyaes
  Building wheel for pyPEG2 (setup.py) ... done
  Created wheel for pyPEG2: filename=pyPEG2-2.15.2-py3-none-any.whl size=22776 sha256=124e6e5d4eab3098fde8c12486ecc64b0a9d76d44aaab200f90ec7964994f43d
  Stored in directory: /root/.cache/pip/wheels/91/8d/f9/39614d868ce42b15c89b7c621bcdd5f63fa84c00d0ca6e037c
  Building wheel for pyaes (setup.py) ... done
  Created wheel for pyaes: filename=pyaes-1.6.1-py3-none-any.whl size=26345 sha256=b3689a7c94c10c19ab7b26fe2339a53954823eb994c10bb847ff18e4986050f5
  Stored in directory: /root/.cache/pip/wheels/0b/7a/53/f1d7837e24ac5d8c0d06b7cbe2144be66dc5e7f37e02f5a164
Successfully built pyPEG2 pyaes
ERROR: duniterpy 0.61.0 has requirement attrs<21.0.0,>=20.2.0, but you'll have attrs 19.1.0 which is incompatible.
```

**Comment by owocki** [comment]
i can upgrade it with `pip install duniterpy --upgrade`

but it updates this dependancy

```
Installing collected packages: attrs
  Attempting uninstall: attrs
    Found existing installation: attrs 19.1.0
    Uninstalling attrs-19.1.0:
      Successfully uninstalled attrs-19.1.0
Successfully installed attrs-20.3.0
``` 

too

File: app/assets/v2/js/pages/profile-trust.js [hunk 7]
```diff
@@ -648,6 +647,114 @@ Vue.component('brightid-verify-modal', {
...
+                        Duniter is a free software for free currencies: a P2P, <a href="https://en.wikipedia.org/wiki/Web_of_trust">Web of Trust</a> & <a href="https://en.wikipedia.org/wiki/Social_credit">Universal Dividend system.</a>
+                        <p class="mb-4">
+                          <a href="https://duniter.org/en/introduction/">Learn more.</a>
+                        </p>
+                      </p>
+                      <div>
+                        <h2 class="font-bigger-4 font-weight-bold"> You need to have these requirements:</h2>
+                        <ol>
+                          <li>let's check if there is already a record with your account.</li>
+                          <li>if there is a link to your gitcoin account in your Duniter record </li>
```
Comment (owocki) [inline app/assets/v2/js/pages/profile-trust.js:684]: i think the user is going to need instructions about how to link their gitcoin account.. like how do i create a record that links my gitcoin/duniter accounts?

File: app/dashboard/views.py (hunk 4 above)
Comment (developerfred) [reply app/dashboard/views.py:3214]: I believe I made a mistake here

**Review by developerfred — COMMENTED** [review]
> ok @developerfred let me know
I'm back I got 20 gigs of space

File: app/assets/v2/js/pages/profile-trust.js (hunk 7 above)
Comment (developerfred) [reply app/assets/v2/js/pages/profile-trust.js:684]: Yes, I will set up a notion with step by step for you to add your Duniter account.

**Comment by owocki** [comment]
thx @developerfred can you turnaround the most recent github comments

**Comment by gdixon** [comment]
Hey @developerfred, thanks for this - we've got travis passing now (we upgraded the ubunutu distro from 14.04 -> 18.04 and installed libsodium-dev). I was wondering if you could help us to understand the full environment requirements for this PR so that we're not pushing this to prod and putting out fires as they happen. 

So far, I know we need:

- libsodium-dev
- openssl >= 1.1.0

Is there anything we should add to this list?

**Comment by developerfred** [comment]
@gdixon only those libs even I had already updated the attrs which was also a requirement.

- [ ] Now I need to add the tutorial for the user to create an account and add his gitcoin address.

**Comment by gdixon** [comment]
That's great, thank you!

What version of attrs did you upgrade from/to? When I tried to upgrade 19.1.0 -> 20.3.0, the tests refused to run?

Could you let us know if you do discover anything else we need to have installed on the host. I'm a little bit worried there might be more hidden requirements that we haven't noticed, but if you say there isn't then I'll trust your judgement.

So, the only packages we need to install before deploy (outside of those handled by python) are:

- libsodium-dev
- openssl >= 1.1.0

Could you also double check https://github.com/gitcoinco/web/pull/8569 - we've added some comments there, it would be awesome to get your input 😄 Thanks again!

**Comment by owocki** [comment]
@developerfred can you pls take a pass here?  waiting on you to get https://github.com/gitcoinco/web/pull/8569 live

**Review by anansiy — APPROVED** [review]

File: app/assets/v2/js/pages/profile-trust.js (hunk 7 above)
Comment (developerfred) [reply app/assets/v2/js/pages/profile-trust.js:684]: https://docs.google.com/document/d/1XhtOELj_YnkkqNsI02ZJHoVh2EDz7L_KvxTNXeiF8Fw/edit?usp=sharing

**Comment by owocki** [comment]
@developerfred hows it going?  let us know if youre blocked on anything

**Comment by developerfred** [comment]
> @developerfred hows it going? let us know if youre blocked on anything

I am blocked in asyncio when calling the view returns 500 error, I think I will have to consume otherwise

**Comment by owocki** [comment]
hmmm how can we consume otherwise? 

if we cant figure it out, can we just back out the actual guts of the duntier integration and have a 'coming soon' in the UI. we can then release the ENS/Facebook/Popp stuff which is held up by being bundled with this.  From there we do a follow up PR when its figured it out.

**Comment by developerfred** [comment]
> hmmm how can we consume otherwise?
> 
> if we cant figure it out, can we just back out the actual guts of the duntier integration and have a 'coming soon' in the UI. we can then release the ENS/Facebook/Popp stuff which is held up by being bundled with this. From there we do a follow up PR when its figured it out.

It's fine by me.

**Comment by owocki** [comment]
cool can you update the PR so its passing + we can do a follow up PR?

**Comment by developerfred** [comment]
> cool can you update the PR so its passing + we can do a follow up PR?

Yes, I’m going to do an isolated job to see what’s going on.

**Comment by owocki** [comment]
@developerfred can you do this this week pls? if not, ill need to figure something else out.

**Comment by developerfred** [comment]
> @developerfred

Yes o/ This is on my roadmap for this week, I will notify you here.

**Comment by thelostone-mc** [comment]
@developerfred I've merged the other PR and disabled the duniter bonus!
If you are planning to wrap that up -> could you update this PR 

Let me know if you cannot get to it and I'll close this out

**Pull request closed without merging**